### DISCUSSION TITLE
Copilot X: "Sorry, but I can only assist with programming related questions."

### DISCUSSION THREAD

**Discussion by mcbaya** [community/community#58923] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I have a frequent and very frustrating problem that, in response to programming questions, Copilot X will respond: "Sorry, but I can only assist with programming related questions."

This happens even in the middle of a conversation about that problem, and as response to questions that Copilot itself has suggested, like "How can I read input from a potentiometer in Arduino?" as in the screenshot below.

In that example, Copilot X had just given me some correct C code for Arduino, but suddenly insisted it couldn't help.

This doesn't seem to be isolated to specific language -- it happens mostly for me in C#.

![image](https://github.com/community/community/assets/99269162/3dc03801-99e9-4e09-b1e3-4b264d32b4a2)

**Comment by bocodez** [comment]
I get this every day and think this feature should be removed. It is annoying to have it thrown into a complex dev discussion stopping the flow of work like a freight train crash. I have to sit and argue with copilot that the question is indeed about programming, and that they can see that it is related to the selected code. Does it really matter if someone trolls copilot into writing some fanfic or talking about their ex? I assume the other safety systems would still operate.

**Comment by kenvix** [comment]
I encountered the same problem when I asked Copilot to `Implement a PWM control code that can blink an LED at 2Hz`. 
It's quite annoying that CopilotX always thinks it's not a programming-related question.

**Comment by mikes-bowden** [comment]
Just canceled my subscription due to this. Every other question or comment is met with, sorry, this isn't programming-related; I can't help. Let me stop what I'm doing and explain why it's programming-related. Oh, in that case... So damn frustrating.

**Comment by itzjoe27** [comment]
I get that issue alot, you can litterally just reply with "It is" and co-pilot will realise it's mistake.

**Comment by mifi** [comment]
can confirm, replying with "it is a programming related question." causes it to reply properly

**Comment by duwejeferson** [comment]
Indeed, arguing with it "resolves" the issue, but it shouldn't be needed.
